### ISSUE TITLE
don't execute asyncData in the vue router beforeEach。

### ISSUE THREAD

**Issue by gotoin** [clarkdo/hare#346]
### What problem does this feature solve?

cache !
i need realize forward refresh and backward do not refresh

now the asyncData  is called every time before loading the component。

i need a configuration that sets the asyncData life cycle

thanks

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/hare/issues/c54">#c54</a>)</em></sub></div>

**Comment by renoirb** [comment]
Oh. Good call.
I've had similar issues to you and couldn't figure out what caused it.

I wonder what would be a better way.

PR?

**Comment by clarkdo** [comment]
@gotoin @renoirb 
If I understood correctly, you can combine the `asyncData` with `store`, only load data in `asyncData` when `store` is null, so that `asyncData` won't be called in backward.

**Comment by gotoin** [comment]
@clarkdo 
Yes, but i need implement a common solution, don't need to set each page, 
as
 page A -> B -> A -> go back(-1)
1. A -> B , push A vnode cache, get B data from asyncData
2. B -> A,  push B vnode cache, get A data from asyncData
3. go back(-1), pop B vnode from cache, then get A vnode

https://github.com/nuxt/nuxt.js/issues/3541
